src/pages/home/report/ReportActionsView.js

```diff
@@ -57,288 +58,252 @@ const defaultProps = {
+        }
+    }, [props.report, props.reportActions, props.network, newMarkerReportActionID]);
 
+    useEffect(() => {
+        const prevReport = prevReportRef.current;
         // Checks to see if a report comment has been manually "marked as unread". All other times when the lastReadTime
         // changes it will be because we marked the entire report as read.
-        const didManuallyMarkReportAsUnread = prevProps.report.lastReadTime !== this.props.report.lastReadTime && ReportUtils.isUnread(this.props.report);
+        const didManuallyMarkReportAsUnread = prevReport && prevReport.lastReadTime !== props.report.lastReadTime && ReportUtils.isUnread(props.report);
         if (didManuallyMarkReportAsUnread) {
-            this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.props.reportActions)});
+            setNewMarkerReportActionID(ReportUtils.getNewMarkerReportActionID(props.report, props.reportActions));
         }
+        // update ref with current report
+        prevReportRef.current = props.report;
+    }, [props.report, props.reportActions]);
 
+    useEffect(() => {
         // Ensures subscription event succeeds when the report/workspace room is created optimistically.
         // Check if the optimistic `OpenReport` or `AddWorkspaceRoom` has succeeded by confirming
         // any `pendingFields.createChat` or `pendingFields.addWorkspaceRoom` fields are set to null.
         // Existing reports created will have empty fields for `pendingFields`.
-        const didCreateReportSuccessfully = !this.props.report.pendingFields || (!this.props.report.pendingFields.addWorkspaceRoom && !this.props.report.pendingFields.createChat);
-        if (!this.didSubscribeToReportTypingEvents && didCreateReportSuccessfully) {
-            Report.subscribeToReportTypingEvents(this.props.report.reportID);
-            this.didSubscribeToReportTypingEvents = true;
-        }
-    }
-
-    componentWillUnmount() {
-        if (this.unsubscribeVisibilityListener) {
-            this.unsubscribeVisibilityListener();
-        }
-
-        if (this.unsubscribeFromNewActionEvent) {
-            this.unsubscribeFromNewActionEvent();
-        }
-
-        Report.unsubscribeFromReportChannel(this.props.report.reportID);
-    }
-
-    /**
-     * @returns {Boolean}
-     */
-    isReportFullyVisible() {
-        return getIsReportFullyVisible(this.props.isFocused);
-    }
-
-    // If the report is optimistic (AKA not yet created) we don't need to call openReport again
-    openReportIfNecessary() {
-        if (this.props.report.isOptimisticReport) {
-            return;
+        const didCreateReportSuccessfully = !props.report.pendingFields || (!props.report.pendingFields.addWorkspaceRoom && !props.report.pendingFields.createChat);
+        if (!didSubscribeToReportTypingEvents && didCreateReportSuccessfully) {
```

Comment:
Coming from https://github.com/Expensify/App/issues/22426#issuecomment-1625449483:
```suggestion
        if (!didSubscribeToReportTypingEvents.current && didCreateReportSuccessfully) {
```